packages/clerk-js/src/v4/UserProfile/AddAuthenticatorApp.tsx

```diff
@@ -0,0 +1,109 @@
+import { TOTPResource } from '@clerk/types';
+import { QRCodeSVG } from 'qrcode.react';
+import React from 'react';
+
+import { useCoreUser } from '../../ui/contexts';
+import { Button, Col, Spinner, Text } from '../customizables';
+import { ClipboardInput, useCardState } from '../elements';
+import { handleError } from '../utils';
+import { FormButtonContainer } from './FormButtons';
+import { NavigateToFlowStartButton } from './NavigateToFlowStartButton';
+import { ContentPage } from './Page';
+
+type AddAuthenticatorAppProps = {
+  title: string;
+  onContinue: () => void;
+};
+
+type DisplayFormat = 'qr' | 'uri';
+
+export const AddAuthenticatorApp = (props: AddAuthenticatorAppProps) => {
+  const { title, onContinue } = props;
+  const user = useCoreUser();
+  const card = useCardState();
+  const [totp, setTOTP] = React.useState<TOTPResource | undefined>(undefined);
+  const [displayFormat, setDisplayFormat] = React.useState<DisplayFormat>('qr');
+
+  React.useEffect(() => {
+    void user
+      .createTOTP()
+      .then((totp: TOTPResource) => setTOTP(totp))
+      .catch(err => handleError(err, [], card.setError));
+  }, [user]);
+
+  if (!totp && !card.error) {
```

Comment:
Did we reach a verdict regarding the position of the spinner? I feel the UX is better if the spinner is within the card, even if an error occurs afterward 

We can totally do this on a different PR

Reply:
No strong opinion, will move inside card